src/build/dotnet/steps/ef-migration-idempotent-script.steps.yaml

```diff
@@ -20,7 +20,14 @@ steps:
     displayName: .NET Restore
     inputs:
       command: restore
-      projects: ${{parameters.efProjectPath}}
+      projects: ${{parameters.defaultProject}}
```

Comment:
I think we'd said to remove the restore steps from this entirely, as they were never in the original and this is not the recommended way of doing EF migrations going forward.

Reply:
As is, this doesn't work for Persimmon without it since adding Audacia.Seed, as it tries to build the projects linked to it. Is there an alternative way?

Reply:
Ok. I'm not sure what impact the addition of Audacia.Seed is having (can you shed any light @OwenLacey28?). As an alternative, because this file is exposing a set of steps, you can add the restore as a step in your job in the Persimmon repo before calling into the template steps (if that makes sense).

Reply:
It causes the script build step to fail. With it being in PS I don't get the error but I assume its the same as the one that happens in the api steps - because we pass in the API project to use as the start up project, and that depends on the Domain project, which has this package
error NU1101: Unable to find package Audacia.Seed.EntityFrameworkCore. No packages exist with this id in source(s): C:\Program Files\dotnet\library-packs, Microsoft Visual Studio Offline Packages, nuget.org

I can call the restore step outside of the Audacia.Build steps, but anyone else would also have the same issue so it seems suited to have in this package? I've updated in line with Jack's suggestion for the other file to combine the restore steps and only happen if using a startup project

Reply:
This method of creating migrations is no longer our recommended method, there was actually discussion on fully removing this file. Therefore I think we should not be changing how it works as it is essentially deprecated and should just be here to ease migration to GitHub for build templates.

Looking at your project pipelines, there seems to have been a structure change to the jobs and steps which is why this build is now failing - we can discuss this further in a chat.

It would be good to remove the restore step entirely from this file as it wasn't there in the original Audacia.Build. (I should not have added this when moving the build files to GitHub)

Reply:
The only structural changes to the project were made after the addition of Audacia.Seed meant calling the private Audacia.Build no longer worked (as you say, it did not have a restore), so the steps were replicated directly in our pipeline instead.

Reply:
RE the package restore issues, I believe this is because the build is only restoring/building the non-test projects. So when it comes to running the tests, it needs to restore/build the test projects too. I believe the same issue would happen if any package hosted on Audacia's public nuget feed were used. I'm not sure how this relates to building the SQL script, but thought I'd offer a bit of context.

**Proposed solution:**
If we're not going to support idempotent scripts in Audacia.Build (which I'm on board with) - why don't we remove it from the repo altogether, and @Rachel-Griffiths-Audacia copies this YAML file down, and we can then work together on a solution that works for her? We can then raise tech debt for moving to a EF Migration bundles (or equivalent)

Reply:
Yes the problem with the building of the EF Script is because there are references to packages such as Audacia.DataAccess and Audacia.Core. This failing has nothing to do with the tests project referencing Audacia.Seed as it will not be trying to build the tests project, but yes, it is the same issue that it cannot restore packages from the Audacia Public feed.

The pipeline was changed, the EF Script was built as part of the Build API job, it has since been moved to its own job which is why there is no restore step.

**RE Proposed Solution**
I think yes, we should remove this file at some point, but I think it will ease migrating to GitHub if we leave it here for now.

Reply:
I've looked with Owen, and can confirm that the restore step is not needed in this template as long as its called from the same job as the asp-core template that has already done it (as you've just commented - I hadn't noticed I'd amended the pipeline in that way when moving to the local script)

I've updated the PR to remove it.

I'm happy with the overall plan being to move to migration bundles, but that would be an additional step for this project - I agree it would ease migrations if both files remained available until all projects changed and some deprecation timeline was followed